lcc_lokavaluto_app_connection/models/debit_request.py

```diff
@@ -75,15 +86,44 @@ def unlink(self):
     def compute_state(self):
         for request in self:
             if not request.debit_move_id:
+                request.state = "draft"
+                continue
+
+            if not request.commission_move_id:
+                # If there is no commission invoice, the debit invoice status defines the request status.
+                request.state = self._convert_status(request.debit_move_state)
                 continue
-            if request.debit_move_state == "draft":
-                request.state = "received"
-            if request.debit_move_state == "posted":
-                request.state = "invoiced"
-            if request.debit_move_state == "paid":
-                request.state = "paid"
-            if request.debit_move_id.state == "cancelled":
+
+            if request.debit_move_state == "cancelled":
+                # Debit move cancellation means there is no debit to perform, then all the process is cancelled.
                 request.state = "cancelled"
+                continue
+
+            if request.commission_move_state == "cancelled":
+                # If the commission invoice is cancelled, we do not consider it anymore for the request status.
+                request.state = self._convert_status(request.debit_move_state)
+                continue
+
+            if request.debit_move_state == "draft" or request.commission_move_state == "draft":
+                # If any of the invoices is in Draft, the request is still considered as "received"
+                request.state = self._convert_status("draft")
+                continue
+
+            if request.debit_move_state == "posted" or request.commission_move_state == "posted":
+                request.state = self._convert_status("posted")
+                continue
+
+            if request.debit_move_state == "paid" or request.commission_move_state == "paid":
+                request.state = self._convert_status("paid")
```

Comment:
I'm a little lost into the intent here, and fear that it's a little too brittle.